cmd/metacache-set.go

```diff
@@ -157,10 +161,27 @@ func (o *listPathOptions) gatherResults(in <-chan metaCacheEntry) func() (metaCa
 				continue
 			}
 			o.debugln("gather got:", entry.name)
-			if o.Marker != "" && entry.name <= o.Marker {
+			if o.Marker != "" && entry.name <= o.Marker && o.VersionMarker == "" {
 				o.debugln("pre marker")
 				continue
 			}
+
+			entry.versionIdx = -1
+			if o.Marker != "" && entry.name <= o.Marker && o.VersionMarker != "" {
+				var xlMeta xlMetaV2
+				idx, err := xlMeta.FindNextVersion(entry.metadata, o.VersionMarker)
+				if err != nil {
+					o.debugln(fmt.Sprintf("unreadable object meta? %s/%s %v", o.Bucket, entry.name, err))
+					continue
+				}
+				if idx == -1 {
+					o.debugln("pre marker and version marker not found")
+					continue
+				}
+				entry.versionIdx = idx
+				o.VersionMarker = ""
```

Comment:
Can you add a comment  to say: "avoid setting VersionMarker in the next key name" ?

Reply:
sure